lib/verdict/experiment.rb

```diff
@@ -274,4 +249,12 @@ def ensure_experiment_has_started
   rescue Verdict::StorageError
     @started_at ||= Time.now.utc
   end
+
+  def fetch_assignment(subject)
+    @storage.retrieve_assignment(self, subject)
+  end
+
+  def fetch_subject(subject_identifier)
```

Comment:
i like the idea of `lookup` calling `fetch_subject`, because it adds a point of "extensibility", but i feel weird about both versions being public.

what do you think? I feel we should make it protected.